src/slice/mealsSlice.tsx

```diff
@@ -65,6 +86,24 @@ export const mealsSlice = createSlice({
       state.snacks = meals.snacks;
       state.supper = meals.supper;
     });
+    builder.addCase(addProduct.fulfilled, (state, action) => {
+      const meals = action.payload;
+      state.date = meals.date;
+      state.breakfast = meals.breakfast;
+      state.dinner = meals.dinner;
+      state.lunch = meals.lunch;
+      state.snacks = meals.snacks;
+      state.supper = meals.supper;
+    });
+    builder.addCase(deleteProduct.fulfilled, (state, action) => {
+      const meals = action.payload;
+      state.date = meals.date;
+      state.breakfast = meals.breakfast;
+      state.dinner = meals.dinner;
+      state.lunch = meals.lunch;
+      state.snacks = meals.snacks;
```

Comment:
Try to create function to avoid duplication (DRY). Can't you just assign action.payload to state? If not then try [fetchMeals, addProduct, deleteProduct].forEach(reducer => builder.addCase(reducer.fulfilled, (state, action) => {
      const meals = action.payload;
      state.date = meals.date;
      state.breakfast = meals.breakfast;
      state.dinner = meals.dinner;
      state.lunch = meals.lunch;
      state.snacks = meals.snacks;
      state.supper = meals.supper;
    });)

Reply:
Done